python/langsmith/_internal/_background_thread.py

```diff
@@ -214,6 +217,23 @@ def tracing_control_thread_func(client_ref: weakref.ref[Client]) -> None:
     scale_up_qsize_trigger: int = batch_ingest_config["scale_up_qsize_trigger"]
     use_multipart = batch_ingest_config.get("use_multipart_endpoint", False)
 
+    if ls_utils.get_env_var("DISABLE_RUN_COMPRESSION") is None and use_multipart:
+        if not (client.info.instance_flags or {}).get(
+            "zstd_compression_enabled", False
+        ):
+            logger.warning(
+                "Zstd compression is not enabled. Please update to the latest "
```

Comment:
nit: "Run compression" instead of "Zstd compression" -- no need to share impl details

Reply:
Would also state in this message that they are falling back to regular multipart